ckan/logic/action/update.py

```diff
@@ -440,6 +438,17 @@ def package_update(
 
             item.after_dataset_update(context, data)
 
+    return_id_only = context.get('return_id_only', False)
+
+    if return_id_only and context.get('defer_commit'):
+        return pkg.id
+
+    pkg_default, pkg_custom = logic.package_show_default_and_custom_schemas(
```

Comment:
This is problematic when code has a need to skip `package_show` and go directly to the model, eg https://github.com/ckan/ckan/issues/3245

How can that code now retrieve the two dict versions?